Source/PercentageGradient.swift

```diff
@@ -7,19 +7,55 @@ public enum GradientDirection {
 
 }
 
+public struct GradientPoints {
+    
+    public var start: CGPoint
+    public var end: CGPoint
+    
+}
+
+public enum GradientAngle {
+    
+    case diagonalAscending
+    case diagonalDescending
+    case horizontal
+    case vertical
+    
+    func gradientPoints() -> GradientPoints {
+        switch self {
+        case .diagonalAscending:
+            return GradientPoints(start: CGPoint(x: 0.0, y: 1.0), end: CGPoint(x: 1.0, y: 0.0))
+        case .diagonalDescending:
+            return GradientPoints(start: CGPoint(x: 0.0, y: 0.0), end: CGPoint(x: 1.0, y: 1.0))
+        case .horizontal:
+            return GradientPoints(start: CGPoint(x: 0.0, y: 0.5), end: CGPoint(x: 1.0, y: 0.5))
+        case .vertical:
+            return GradientPoints(start: CGPoint(x: 0.5, y: 0.0), end: CGPoint(x: 0.5, y: 1.0))
+        }
+    }
+    
+}
+
 public struct PercentageGradient: Gradient {
 
+    public var angle: GradientAngle?
```

Comment:
This seems like a good addition, but I'm not sure it should be optional. If we're going to apply the concept of angles to all the gradients, then I think it should be a required property (and let rather than var). What do you think?